### PR TITLE
Fix build for Python 3 on OS X

### LINKED ISSUE #197: pip install vmprof failed, duplicated symbol
/usr/bin/clang -bundle -undefined dynamic_lookup -arch i386 -arch x86_64 -g build/temp.macosx-10.6-intel-3.6/src/_vmprof.o build/temp.macosx-10.6-intel-3.6/src/machine.o build/temp.macosx-10.6-intel-3.6/src/compat.o build/temp.macosx-10.6-intel-3.6/src/vmp_stack.o build/temp.macosx-10.6-intel-3.6/src/vmprof_common.o build/temp.macosx-10.6-intel-3.6/src/vmprof_memory.o build/temp.macosx-10.6-intel-3.6/src/symboltable.o build/temp.macosx-10.6-intel-3.6/src/vmprof_unix.o build/temp.macosx-10.6-intel-3.6/src/vmprof_mt.o -o build/lib.macosx-10.6-intel-3.6/_vmprof.cpython-36m-darwin.so

ld: warning: The i386 architecture is deprecated for macOS (remove from the Xcode build setting: ARCHS)
    ld: warning: ignoring file /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/lib/libSystem.tbd, missing required architecture i386 in file /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/lib/libSystem.tbd
    **duplicate symbol __PyThreadState_Current in:
        build/temp.macosx-10.6-intel-3.6/src/_vmprof.o
        build/temp.macosx-10.6-intel-3.6/src/vmprof_common.o
    duplicate symbol __PyThreadState_Current in:
        build/temp.macosx-10.6-intel-3.6/src/_vmprof.o
        build/temp.macosx-10.6-intel-3.6/src/vmprof_unix.o**
    ld: 2 duplicate symbols for architecture i386
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    error: command '/usr/bin/clang' failed with exit status 1

### PR DESCRIPTION
**Pull request by timpalpant**
vmprof_common.h gets included on OS X, resulting in multiple definitions of `_PyThreadState_Current`. This definition is only needed in `vmprof_win.c`, so move it there.

I pip installed and ran pytest on the platforms I have available (OS X and Linux). Unfortunately, I don't have access to a Windows machine to test it.

Fixes #197 and #189

### CODE REVIEW

**Comment by om26er** [comment]
@timpalpant Could you rebase this on top of master and see if the CI likes that ?

**Comment by NyanKiyoshi** [comment]
@timpalpant could you try to rebase whenever you have a second?

**Comment by mattip** [comment]
Any thoughts why two of the windows builds are failing?

File: src/vmprof_win.c [hunk 1]
```diff
@@ -104,6 +104,17 @@ int vmprof_snapshot_thread(DWORD thread_id, PY_WIN_THREAD_STATE *tstate, prof_st
...
+   to do that without these hacks.  We can't use PyThreadState_GET(),
+   because that calls PyThreadState_Get() which fails an assert if the
+   result is NULL. */
+#if PY_MAJOR_VERSION >= 3 && !defined(_Py_atomic_load_relaxed)
+                             /* this was abruptly un-defined in 3.5.1 */
+void *volatile _PyThreadState_Current;
+   /* XXX simple volatile access is assumed atomic */
+#  define _Py_atomic_load_relaxed(pp)  (*(pp))
+#endif
+
```
Comment (NyanKiyoshi) [inline src/vmprof_win.c]: Seems like it might be supposed be inside `vmprof_win.h` instead of `vmprof_win.c` has it is used by `_vmprof.c`?

**Comment by NyanKiyoshi** [comment]
@mattip seems like master has the same issues, so just ignore it https://ci.appveyor.com/project/planrich/vmprof-python/builds/31240621

**Comment by mattip** [comment]
thanks, merged

**Pull request merged**